### PR TITLE
Preversioning; Packaging and CLI bug fixes

### LINKED ISSUE #533: Summary of unresolved packaging issues
Right now, packaging doesn't actually work: if one tries to `pip install git+https://github.com/nucypher/nucypher.git`, it will result in unworkable nucypher installation, and only "editable" installation works. Here's the summary of things to fix that:
* Install nucypher submodules. Currently `setup.py` has only `packages=[PACKAGE_NAME, 'cli']`. In fact, there are many more which currently aren't installed (unless the installation is editable, so they are already there). Usually `find_modules()` function from `setuptools` is used to get the full package list for this keyword argument.
* `tests` directory is a module, so it will appear in `find_modules()`. We don't want `tests` to be a module. If `__init__.py` is removed from `tests`, and `from .fixtures import *` is replaced with `from fixtures import *`, it works (yes, even in Pyhon3).
* `cli` is a global module. That's a very generic name which can clash with something. Actually,  `cli/main.py` can be moved to `nucypher/cli.py`. Alternatively, it can live in a namespace separate from `nucypher`, but I don't see a point of it. Entry point in `setup.py` should be changed accordingly.
* Finally, CircleCI config should be changed so that it installs `nucypher` as non-editable. This will make a more realistic test which will break down if our packaging somehow stops working.

### PR DESCRIPTION
**Pull request by KPrasch**
This PR makes nucypher installable via pip with the proper package metadata, data files, up-to-date dependency specifications, and CLI entry points.

* Introduces `nucypher configure forget`
* More informative exception handing
* Various bug fixes

Fixes #533 #502  #520  #522 #517 #518 #529, #481, #454  and #497

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -137,31 +138,38 @@ def run(self):
...
+          'blockchain/eth/sol/source/contracts/*',
+          'blockchain/eth/sol/source/contracts/lib/*',
+          'blockchain/eth/sol/source/contracts/proxy/*',
+          'blockchain/eth/sol/source/zepellin/math/*',
+          'blockchain/eth/sol/source/zepellin/ownership/*',
+          'blockchain/eth/sol/source/zepellin/token/*']},
       include_package_data=True,
-      entry_points='[console_scripts]\n{}=cli.main:cli'.format(PACKAGE_NAME),
+
+      entry_points={'console_scripts': ['{0}={0}.cli:cli'.format(PACKAGE_NAME)]},
```
Comment (michwill) [inline setup.py]: Well, again. Would that ever work with non-editable installation?

Comment (michwill) [reply setup.py]: (since it's not a module)

Comment (KPrasch) [reply setup.py]: To clarify - I think this is acceptable for a developer installation - I went back-and-forth for a while on where to put this file, but ultimately inside `nucypher` is the easiest place to manage it. 

Are you able to pip install the project directory? `pip install ./nucypher`

Comment (michwill) [reply setup.py]: Ah wait, you moved it to nucypher/cli.py. Perfect!

Comment (michwill) [reply setup.py]: Can test like:
```
virtualenv .test -p python3
source .test/bin/activate
pip3 install git+https://github.com/KPrasch/nucypher.git@preversioning
nucypher
```

**Pull request merged**